### ISSUE TITLE
Refactor Blockchain.getBlockHeader()

### ISSUE THREAD

**Issue by christroutner** [Bitcoin-com/bitbox-javascript-sdk#58]
This Issue is tagged with a bounty of **0.4 BCH**. Once a Pull Request (PR) has been verified to meet the acceptance criteria, and merged into the [ct-v2-refactor branch](https://github.com/Bitcoin-com/bitbox-javascript-sdk/pull/52), the funds will be released to the developer.

## Overview
rest.bitcoin.com is a web service running as a REST API. It communicates with a Bitcoin Cash full node, Insight API, BitDB database, and a growing array of other bitcoin-focused databases. BITBOX JavaScript SDK is an [npm library](https://www.npmjs.com/package/bitbox-sdk) used by developers to leverage the rest.bitcoin.com web service. [Learn More](https://developer.bitcoin.com/bitbox/docs/getting-started)

rest.bitcoin.com is wrapping up a large refactor, referred to as V2. BITBOX SDK is currently being refactored to reflect the new endpoints. The scope of this Issue is to refactor the BITBOX `Blockchain.getBlockHeader()` endpoint to use the new GET and POST endpoints in rest.bitcoin.com.

[This is the existing code that needs to be refactored](https://github.com/Bitcoin-com/bitbox-javascript-sdk/blob/a20225afce76dfbcc018e72d148182fac59cdc84/src/Blockchain.js#L69-L81)

Before starting work on this Issue, it is advisable to add a comment to this issue with a link to your fork of the code base, so that others can follow the progress of your work. If work pauses for more than 48 hours, other developers are encouraged to pick up this Issue. 

Questions or concerns about this Issue or bounty can be left in the comments of this Issue, or discussed on our [developer Discord channel](https://discord.gg/m5h28ND).

## Notes
- The code governing the control of the endpoint on the rest.bitcoin.com side [can be viewed here](https://github.com/Bitcoin-com/rest.bitcoin.com/blob/55b32064bdd81a43e6068ab8e9e2c10b8fce68c2/src/routes/v2/blockchain.ts#L237-L359).

- The [Address](https://github.com/Bitcoin-com/bitbox-javascript-sdk/blob/34d98a1f2dc1733652e1e4a03b58f93276af923f/src/Address.js#L269-L298) and [Block](https://github.com/Bitcoin-com/bitbox-javascript-sdk/blob/34d98a1f2dc1733652e1e4a03b58f93276af923f/src/Block.js#L7-L35) endpoints have already been refactored in a manner similar to the specifications in this Issue. These endpoints can be used as an example of the code changes required by this Issue.

## Specifications
The PR submitted to fulfill this Issue should have the following features:
- The [existing](https://github.com/Bitcoin-com/bitbox-javascript-sdk/blob/master/src/Blockchain.js#L69-L81) `getBlockHeader()` function needs to be refactored according to the following criteria:
  - An input `string` should trigger a GET call `rest.bitcoin.com/v2/blockchain/getBlockHeader`. The `verbose` query string should also be transmitted and appropriately control the output.

  - An input `array` of strings should trigger a POST call to `rest.bitcoin.com/v2/blockchain/getBlockHeader`. The `verbose` input should also be passed along as part of the body.

  - An input not matching either of the above conditions should throw an error.

- Passing integration tests must be included to cover the above code changes. [Here is an example](https://github.com/Bitcoin-com/bitbox-javascript-sdk/blob/34d98a1f2dc1733652e1e4a03b58f93276af923f/test/integration/address.js#L25-L96) of integration tests for similar code changes to the `Address.details` endpoint.

## Acceptance Criteria
In order for a developer to receive the bounty attached to this issue, they must meet the following acceptance criteria:

- A PR must be submitted to this repository with the code changes listed in the Specifications section.
- All unit tests must pass.
- All integration tests must pass.

**Comment by christroutner** [comment]
Nice! Thank you @crazymerlyn! I'll review this tomorrow.

**Comment by christroutner** [comment]
Closeing this issue. The bounty has been fulfilled by PR #61.
